src/python/tensorflow_cloud/tuner/tuner.py

```diff
@@ -0,0 +1,358 @@
+# Lint as: python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""KerasTuner CloudOracle and CloudTuner classes."""
+
+import datetime
+import time
+from typing import Any, Callable, Dict, List, Mapping, Optional, Text, Union
+
+from kerastuner.engine import hypermodel as hypermodel_module
+from kerastuner.engine import hyperparameters as hp_module
+from kerastuner.engine import oracle as oracle_module
+from kerastuner.engine import trial as trial_module
+from kerastuner.engine import tuner as tuner_module
+import tensorflow as tf  # pylint: disable=g-explicit-tensorflow-version-import
+
+from tensorflow_cloud.tuner import utils
+from tensorflow_cloud.tuner import optimizer_client
+
+
+class CloudOracle(oracle_module.Oracle):
+    """KerasTuner Oracle interface implemented by talking to CAIP Optimizer Service."""
+
+    def __init__(
+        self,
+        project_id,
+        region,
+        objective=None,
+        hyperparameters=None,
+        study_config=None,
+        max_trials=None,
+        study_id=None,
+    ):
+        """KerasTuner Oracle interface implmemented by talking to CAIP Optimizer Service.
```

Comment:
Is this indentation correct? Here and below.

Reply:
good catch, I applied black formatting as current github actions on tf/cloud enforce flake8 ... however like black does not support  4 indentation and it does not handle updating the doc string indentation properly. I will update the indents manually.

Reply:
Done